doc/user-guide/Push-notifications.md

```diff
@@ -0,0 +1,205 @@
+## Push Notifications
+
+MongooseIM server supports push notifications using FCM (**F**irebase **C**loud **M**essaging)
+and APNS (**A**pple **P**ush **N**otification **S**ervice) providers. Server side push
+notification support is fully compliment with [XEP-0357 Push Notifications][XEP-0357], which defines several components that need to work together in order to provide clients with working push notifications.
+The following list shows those components as defined in
+[XEP-0357][] and MongooseIM components that correspond to those entities:
+
+  * _XMPP Server_ in MongooseIM is enabled by module [mod_push][]
+  * _App Server_ in MongooseIM is enabled by adding `node_push` node type to [mod_pubsub][]'s configuration
+  * _XMPP Push Service_ is implemented as [MongoosePush][] application
+
+All those entities have to be enabled and properly configured in order to use push notifications.
+So let's get it, shall we?
+
+### Overall component architecture
+
+The overall architecture of all components that make push notifications possible in MongooseIM, looks as follows:
+![](push_notifications_components.svg)
+
+Please note that even that, there are two domains of MongooseIM and third one for [MongoosePush][], it is not required. All those 3 components can be on the same host with the same domain. 
+
+### Configuring MongooseIM components
+
+Firstly, let's configure all required MongooseIM components, step by step.
+
+#### [mod_push][] a.k.a. '_XMPP Server_'
+
+The first component that we need to configure in MongooseIM is [mod_push][] module.
+This module directly communicates with XMPP clients in order to enable/disable notifications on per-client basis.
+
+The [mod_push][] module is very easy to enable - just paste in your MongooseIM configuration file:
+
+ ```Erlang
```

Comment:
This code is rendered badly on readthedocs. Should the language start with lowercase?